src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -166,22 +181,14 @@ class AuthScreens extends React.Component {
     }
 
     /**
-     * @param {String} [url]
-     * @returns {Boolean}
+     * Navigate to the transition exit route
+     *
+     * @param {String} exitTo
      */
-    shouldCreateFreePolicy(url = '') {
-        if (!url) {
-            return false;
-        }
-
-        const path = new URL(url).pathname;
-        const params = new URLSearchParams(url);
-        const exitTo = params.get('exitTo');
-        const email = params.get('email');
-        const isLoggingInAsNewUser = !_.isNull(this.props.session.email) && (email !== this.props.session.email);
-        return !isLoggingInAsNewUser
-            && Str.startsWith(path, Str.normalizeUrl(ROUTES.LOGIN_WITH_SHORT_LIVED_TOKEN))
-            && exitTo === ROUTES.WORKSPACE_NEW;
+    navigateToExitRoute(exitTo) {
```

Comment:
Since this is only called once and it only does two things, I don't think it's necessary to make it a method.